src/workerd/util/sqlite-kv.h

```diff
@@ -5,9 +5,23 @@
 #pragma once
 
 #include "sqlite.h"
+#include <workerd/jsg/exception.h>
 
 namespace workerd {
 
+namespace {
+
+// Small class which is used to customeze certain aspects of the underlying sql operations
+// In this case we just customise the error reporting to emit JSG user visible errors instead
+// of KJ exceptions which become internal errors.
+class SqliteKvRegulator: public SqliteDatabase::Regulator {
+  void onError(kj::StringPtr message) const override {
+    JSG_ASSERT(false, Error, message);
```

Comment:
This implementation will throw JSG errors for all possible SQLite errors.  Almost everything about this class is under our control so every error EXCEPT the TOOBIG error should be an internal error.  Is it possible to report only TOOBIG errors this way?

Reply:
Yes, but not without either doing matching on the string message (bad), or changing the onError API (out of scope before birthday week).

What I'll do now is only pass the regulator to the put query, so only sql errors for put get passed through. Or would you prefer I do string matching as a temporary kludge here?

Reply:
Changed it to only report errors on the put query. Let me know your thoughts on that. I acknowledge it's a bit of a hack, but I think its a low consequence change here. Reporting an SQLite error that's actually our fault to the user doesn't seem like a massive issue for sqlite here.

Reply:
Hmm, including the error code in the onError doesn't seem that difficult actually. I'll try to get something out based on that.

Reply:
Pushed a fixup commit passing the error code to onError. Let me know what you think of that, and I'll squash it if you think it's a good approach here.